Sorter.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PhoneBook
+{
+    public delegate bool SortType(Contact cont1, Contact cont2);
+
+    public static class Sorter
+    {
+        public static void SortPhoneBook(SortType del)
+        {
+            MergeSort(ConsoleHost.contacts, 0, ConsoleHost.contacts.Length - 1, del);
+        }
+
+        public static bool AscendingSortByName(Contact cont1, Contact cont2)
+        {
+            if (string.Compare(cont1.Name, cont2.Name) == 0 || string.Compare(cont1.Name, cont2.Name) == -1)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public static bool AscendingSortBySurName(Contact cont1, Contact cont2)
+        {
+            if ((string.Compare(cont1.SurName, cont2.SurName) == 0 || string.Compare(cont1.SurName, cont2.SurName) == -1) &&
+                (cont1.SurName != "") && cont2.SurName != "")
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public static bool AscendingSortByPhone(Contact cont1, Contact cont2)
+        {
+            if (string.Compare(cont1.Phone.Substring(0, 3), cont2.Phone.Substring(0, 3)) == 0 || string.Compare(cont1.Phone.Substring(0, 3), cont2.Phone.Substring(0, 3)) == -1) 
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public static bool DescendingSortByPhone(Contact cont1, Contact cont2)
+        {
+            if (string.Compare(cont1.Phone.Substring(0, 3), cont2.Phone.Substring(0, 3)) == 0 || string.Compare(cont1.Phone.Substring(0, 3), cont2.Phone.Substring(0, 3)) == -1)
```

Comment:
before using `Substring(0, 3)`  make sure it has the appropriate length. Also store `cont1.Phone.Substring(0, 3)` in variable to use it multiple times